AEPAnalytics/Sources/AnalyticsMigrator.swift

```diff
@@ -0,0 +1,98 @@
+/*
+ Copyright 2021 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+import AEPServices
+
+public class AnalyticsMigrator {
+    private static let LOG_TAG = "AnalyticsMigrator"
+
+    private static var userDefaults: UserDefaults {
+        if let appGroup = ServiceProvider.shared.namedKeyValueService.getAppGroup(), !appGroup.isEmpty {
+            return UserDefaults(suiteName: appGroup) ?? UserDefaults.standard
+        }
+
+        return UserDefaults.standard
+    }
+
+    /// Migrate analytics data from v4 local storage
+    /// - Parameters:
+    ///   - dataStore: DataStore to store persisted analytics data
+    private static func migrateFromV4(dataStore: NamedCollectionDataStore) {
+        Log.trace(label: LOG_TAG, "Migration started for Analytics data from v4")
+        if let aid = userDefaults.string(forKey: AnalyticsConstants.V4Migration.AID) {
+            dataStore.set(key: AnalyticsConstants.DataStoreKeys.AID, value: aid)
+        }
+        if let ignoreAid = userDefaults.object(forKey: AnalyticsConstants.V4Migration.IGNORE_AID) {
+            dataStore.set(key: AnalyticsConstants.DataStoreKeys.IGNORE_AID, value: ignoreAid as? Bool)
+        }
+        if let vid = userDefaults.string(forKey: AnalyticsConstants.V4Migration.VID) {
+            dataStore.set(key: AnalyticsConstants.DataStoreKeys.VID, value: vid)
+        }
+
+        userDefaults.removeObject(forKey: AnalyticsConstants.V4Migration.AID)
```

Comment:
Since userDefaults is a computed property. Can we hold on to its reference in beginning of this method.
`let userDefaults = self.userDefaults`